lib/inject.rb

```diff
@@ -0,0 +1,42 @@
+class Array
+
+	def another_inject(*args,&block)
+		copy_of_self = self.clone
+
+		if args[0].is_a? Symbol
+			block = Proc.new { |memo,x| memo.send args[0] , x}
+		elsif args.count == 2
+			block = Proc.new { |memo,x| memo.send args[1] , x}
+			copy_of_self.unshift(args[0])
+		elsif args != []
```

Comment:
`elsif` is usually a code smell... especially seeing as you aren't terminating with an `else`